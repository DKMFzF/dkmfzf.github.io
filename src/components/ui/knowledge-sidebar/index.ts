export { KnowledgeSidebarUI } from './knowledge-sidebar';
