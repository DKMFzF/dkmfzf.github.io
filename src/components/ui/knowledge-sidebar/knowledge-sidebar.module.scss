@use '~styles/variables' as *;
@use '~styles/mixins' as *;

$font-size-footer-card: clamp(0.5rem, -0.05rem + 1.75vw, 0.94rem);

.knowledgeSidebar {
	position: relative;
	display: flex;
	flex-direction: column;
	justify-content: center;
	gap: 20px;
	block-size: 100%;

	&__bodySlider {
		display: flex;
		justify-content: space-between;
		align-items: center;
		width: 100%;
		margin: auto 0;
	}

	&__button {
		display: flex;
		justify-content: center;
		align-items: center;
		padding: 10px;
		color: white;
		background: none;
		border: 1px solid white;
		border-radius: 50%;
		cursor: pointer;
		transition: opacity 0.2s ease;

		&:hover {
			opacity: 0.7;
		}

		&:focus {
			outline: none;
		}

		svg {
			width: 24px;
			height: 24px;
		}
	}

	&__content {
		position: relative;
		display: flex;
		flex: 1;
		justify-content: center;
		align-items: center;
		height: 100%;
		overflow: hidden;
		text-align: center;
	}

	&__card {
		display: flex;
		flex-direction: column;
		justify-content: center;
		width: 100%;
		height: 100%;
		color: white;
		border-radius: $section-border-radius;
	}

	&__skill {
		font-size: 65px;
		font-family: $font-title;
	}

	&__stackDiscription {
		display: flex;
		flex-direction: column;
		justify-content: space-between;
		height: auto;
		min-height: 111px;
		font-size: 15px;
		line-height: 1.4;
		text-transform: uppercase;
	}

	&__stackArrow {
		inline-size: fit-content;
		padding: 3px 6px;
		color: $bg-primary-color;
		background-color: white;
		border-radius: 50px;
	}
}

.fadeIn {
	animation: fade-in 0.3s ease-in-out forwards;
}

.fadeOut {
	animation: fade-out 0.3s ease-in-out forwards;
}

@keyframes fade-in {
	from {
		opacity: 0;
	}

	to {
		opacity: 1;
	}
}

@keyframes fade-out {
	from {
		opacity: 1;
	}

	to {
		opacity: 0;
	}
}
