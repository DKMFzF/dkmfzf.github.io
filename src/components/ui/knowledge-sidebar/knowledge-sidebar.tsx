import { FC } from 'react';
import { knowledgeCards } from '@utils-constants';
import styles from './knowledge-sidebar.module.scss';
import { useCardSlider } from '@hooks';

export const KnowledgeSidebarUI: FC = () => {
	const { currentIndex, transitioning, handleNext, handlePrev } =
		useCardSlider(knowledgeCards.length);

	return (
		<div className={styles['knowledge-sidebar']}>
			<div className={styles['knowledge-sidebar__body-slider']}>
				<button
					onClick={handlePrev}
					className={styles['knowledge-sidebar__button']}
					aria-label='Previous card'
					data-cy='card-btn-left'
				>
					<svg
						width='24'
						height='24'
						viewBox='0 0 24 24'
						fill='none'
						xmlns='http://www.w3.org/2000/svg'
					>
						<path
							d='M15 18L9 12L15 6'
							stroke='currentColor'
							strokeWidth='2'
							strokeLinecap='round'
							strokeLinejoin='round'
						/>
					</svg>
				</button>

				<div
					aria-live='polite'
					className={styles['knowledge-sidebar__content']}
				>
					<div
						className={`${styles['knowledge-sidebar__card']} ${
							transitioning
								? styles['fade-out']
								: styles['fade-in']
						}`}
						key={knowledgeCards[currentIndex].id}
					>
						<span
							className={styles['knowledge-sidebar__skill']}
							data-cy='card-title'
						>
							{knowledgeCards[currentIndex].title}
						</span>
					</div>
				</div>

				<button
					onClick={handleNext}
					className={styles['knowledge-sidebar__button']}
					aria-label='Next card'
					data-cy='card-btn-right'
				>
					<svg
						width='24'
						height='24'
						viewBox='0 0 24 24'
						fill='none'
						xmlns='http://www.w3.org/2000/svg'
					>
						<path
							d='M9 18L15 12L9 6'
							stroke='currentColor'
							strokeWidth='2'
							strokeLinecap='round'
							strokeLinejoin='round'
						/>
					</svg>
				</button>
			</div>

			<footer
				aria-live='polite'
				className={styles['knowledge-sidebar__stack-discription']}
			>
				<span
					className={
						transitioning ? styles['fade-out'] : styles['fade-in']
					}
				>
					{knowledgeCards[currentIndex].content}
				</span>
				<div className={styles['knowledge-sidebar__stack-arrow']}>
					{'>'}
				</div>
			</footer>
		</div>
	);
};
