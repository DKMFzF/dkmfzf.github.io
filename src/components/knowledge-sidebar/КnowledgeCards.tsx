import { FC } from 'react';
import { KnowledgeSidebarUI } from '../ui';

export const KnowledgeSidebar: FC = () => <KnowledgeSidebarUI />;
