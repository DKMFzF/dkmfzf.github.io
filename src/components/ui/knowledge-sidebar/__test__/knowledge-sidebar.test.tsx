import { testRenderComponent } from '@utils-testing';
import { KnowledgeSidebarUI } from '../knowledge-sidebar';

describe('[KnowledgeSidebarUI]', () => {
	it('Renders knowledge-sidebar correctly and matches snapshot', () =>
		testRenderComponent(<KnowledgeSidebarUI />));
});
