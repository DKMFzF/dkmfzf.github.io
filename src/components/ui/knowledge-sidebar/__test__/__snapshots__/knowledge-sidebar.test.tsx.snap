// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[KnowledgeSidebarUI] Renders knowledge-sidebar correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="knowledge-sidebar"
  >
    <div
      class="knowledge-sidebar__body-slider"
    >
      <button
        aria-label="Previous card"
        class="knowledge-sidebar__button"
        data-cy="card-btn-left"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 18L9 12L15 6"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
      <div
        aria-live="polite"
        class="knowledge-sidebar__content"
      >
        <div
          class="knowledge-sidebar__card fade-in"
        >
          <span
            class="knowledge-sidebar__skill"
            data-cy="card-title"
          >
            React
          </span>
        </div>
      </div>
      <button
        aria-label="Next card"
        class="knowledge-sidebar__button"
        data-cy="card-btn-right"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 18L15 12L9 6"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <footer
      aria-live="polite"
      class="knowledge-sidebar__stack-discription"
    >
      <span
        class="fade-in"
      >
        Experience in SPA development. I work confidently with hooks, components, and routing.
      </span>
      <div
        class="knowledge-sidebar__stack-arrow"
      >
        &gt;
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
